clarity/src/vm/ast/mod.rs

```diff
@@ -200,6 +227,19 @@ pub fn inner_build_ast<T: CostTracker>(
         }
         _ => (),
     }
+
+    if ast_rules != ASTRules::Typical {
+        // run extra stack-depth pass for tuples
+        match VaryStackDepthChecker::run_pass(&mut contract_ast, clarity_version) {
+            Err(e) if error_early => return Err(e),
+            Err(e) => {
+                diagnostics.push(e.diagnostic);
+                success = false;
+            }
+            _ => (),
+        }
+    }
```

Comment:
I think this should only need to run in `Epoch205`, right? Parser v2 now catches this problem without the extra check.

Reply:
Stacks 2.1 will need to behave identically to a Stacks 2.05 node until the activation block height.